cloudinit/distros/arch.py

```diff
@@ -43,16 +43,17 @@ def __init__(self, name, cfg, paths):
         cfg['ssh_svcname'] = 'sshd'
 
     def apply_locale(self, locale, out_fn=None):
-        if not out_fn:
-            out_fn = self.locale_conf_fn
-        subp.subp(['locale-gen', '-G', locale], capture=False)
-        # "" provides trailing newline during join
         lines = [
             util.make_header(),
-            'LANG="%s"' % (locale),
+            # Hard-coding the charset isn't ideal, but there is no other way.
+            '%s UTF-8' % (locale),
             "",
         ]
-        util.write_file(out_fn, "\n".join(lines))
+        util.write_file(self.locale_gen_fn, "\n".join(lines))
+        subp.subp(['locale-gen'], capture=False)
+        # In the future systemd can handle locale-gen stuff:
+        # https://github.com/systemd/systemd/pull/9864
+        subp.subp(['localectl', 'set-locale', locale], capture=False)
```

Comment:
What does running `localectl` give us that running the code above it didn't?

Reply:
The old code was broken, the line written to `/etc/locale.gen` was wrong and it didn't write to `/etc/locale.conf`.

Sure, I could just fix the code so it write the correct lines to both `/etc/locale.gen` and `/etc/locale.conf`, but `localectl` also updates the systemd [environment block](https://www.freedesktop.org/software/systemd/man/systemctl.html#Environment%20Commands) and emits a `PropertiesChanged` signal for the `org.freedesktop.locale1` `Locale` property.

... also in the future I want to remove the `locale.gen` logic as systemd can handle it (https://github.com/systemd/systemd/pull/9864), just waiting for it to ship.